### PR TITLE
[3.0] Public Cloud: Add support for bootstrapping in Google Compute

### PR DESCRIPTION
**Pull request by bear454**
(cherry picked from commit 446f412a3fd871bfae8429d9131fe462cbbb0d5e)

### CODE REVIEW

**Comment by bear454** [comment]
[Public Cloud: Add support for bootstrapping in Google Compute](https://trello.com/c/N0jyJj5v/1051-public-cloud-add-support-for-bootstrapping-in-google-compute)

**Comment by MaximilianMeister** [comment]
@bear454 can you make sure to get the tests green here?

**Comment by bear454** [comment]
Sorry @MaximilianMeister - I thought `product_name` was defined for 3.x.

**Comment by MaximilianMeister** [comment]
LGTM :+1: 

@jordimassaguerpla is it ok like this or should we squash the commits or add the bug reference to the 2nd commit as well? in jenkins we test that every backport commit has a bug reference.

@bear454 feel free to squash the commits, or add the bug reference to the 2nd commit

**Comment by MaximilianMeister** [comment]
> I thought product_name was defined for 3.x.

yeah as we havent backported the branding this was not really obvious

**Comment by bear454** [comment]
I've squashed the commit, but there's no bugzilla entry for adding public cloud support, @jordimassaguerpla .

@MaximilianMeister , please reset your review; github removed it when I squashed the commit.

**Comment by jordimassaguerpla** [comment]
@bear454 : fine with me. We will figure out during the packaging process if we have to add some reference.

**Comment by bear454** [comment]
Rebased, and added in the necessary fixes for BSC#1098085, as @kiall requested.

**Comment by bear454** [comment]
[Bug 1098085 - No UI for accessing node bootstrap info after setup](https://trello.com/c/JBsPkG6f/1086-bug-1098085-no-ui-for-accessing-node-bootstrap-info-after-setup)

**Comment by kiall** [comment]
@MaximilianMeister - can you double check? Should be good to go though.

File: app/controllers/setup_controller.rb [hunk 1]
```diff
@@ -11,7 +11,8 @@ class SetupController < ApplicationController
 
   skip_before_action :redirect_to_setup
   before_action :redirect_to_dashboard
-  skip_before_action :redirect_to_dashboard, only: :worker_bootstrap
+  skip_before_action :redirect_to_dashboard,
+                     only: [:worker_bootstrap, :build_cloud_cluster]
```
Comment (MaximilianMeister) [inline app/controllers/setup_controller.rb]: :+1:

**Pull request merged**